src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/IProjectTreeExtensions.cs

```diff
@@ -19,6 +20,20 @@ internal static class IProjectTreeExtensions
                 caption);
         }
 
+        /// <summary>
+        /// Gets the direct child of <paramref name="tree"/> that represents <paramref name="dependency"/> if
+        /// one exists, otherwise <see langword="null"/>.
+        /// </summary>
+        [Pure]
```

Comment:
What does `[Pure]` mean?

Reply:
It's a way of annotating that the function has no side effects when called. In practice it amounts to meaning "you should use this method's return value, otherwise calling this method has no effect."

See also https://github.com/dotnet/runtime/issues/34098